src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -54,6 +88,16 @@
             """)
     List<Map<String, Object>> getExercisesDue(@Param("startDate") ZonedDateTime startDate, @Param("endDate") ZonedDateTime endDate);
 
+    @Query("""
+            select e.dueDate as day, count(e.id) as amount
+            from Exercise e
+            where e.dueDate >= :#{#startDate} and e.dueDate <= :#{#endDate} and e.id in :exerciseIds
+            group by e.dueDate
+            order by e.dueDate asc
+            """)
+    List<Map<String, Object>> getExercisesDueForCourse(@Param("startDate") ZonedDateTime startDate, @Param("endDate") ZonedDateTime endDate,
```

Comment:
please use a custom object here and not a map

Reply:
Done